### ISSUE TITLE
One of the tests in the test suite fails because of a runtime accessibility check.

### ISSUE THREAD

**Issue by dvraaij** [AdaCore/aunit#55]
One of the tests in the test suite fails because of a runtime accessibility check on line 101 of file `aunit-simple_test_cases.adb` (see below). A workaround is to replace all runtime checks for anonymous access types with compile-time checks based on the [designated type model](https://blog.adacore.com/going-beyond-ada-2022) model (`-gnatd_b`).

```raw
$ ./exe/aunit_harness

OK (suite) Add test case (in 0.000003 sec.)
OK (suite) Run empty suite (in 0.000001 sec.)
OK (suite) Run suite with a successful test (in 0.000003 sec.)
OK (suite) Run suite with a failing test (in 0.000078 sec.)
OK (suite) Run suite with a test raising an exception (in 0.000858 sec.)
OK (suite) Run suite with various tests (in 0.000567 sec.)
OK (suite) Verify Set_Up/Tear_Down are called (in 0.000001 sec.)
OK (test_case) Test routines registration (in 0.000002 sec.)
OK (test_case) Test set_up phase (in 0.000000 sec.)
OK (test_case) Test tear_down phase (in 0.000000 sec.)
OK (test_fixture) Test Set_Up call (in 0.000017 sec.)
OK (test_fixture) Test Tear_Down call (the called test is success) (in 0.000001 sec.)
OK (test_fixture) Test Tear_Down call (the called test is failure) (in 0.000011 sec.)
OK (test_fixture) Test Tear_Down call (the called test is error) (in 0.000517 sec.)

ERROR (test_case) Test run phase (in 0.000021 sec.)
    PROGRAM_ERROR
    Exception Message: aunit-simple_test_cases.adb:101 accessibility check failed
    Traceback:
        [/usr/lib64/libaunit.so.24.0.0]
        0x7fe0ca13e72e
        0x7fe0ca13e901
        [./exe/aunit_harness]
        0x407cb1 Aunit.Test_Cases.Tests.Test_Run at aunit-test_cases-tests.adb:66
        0x408efc Aunit.Test_Cases.Tests.Suite.Caller.Run_TestXnnn at aunit-test_caller.adb:96
        [/usr/lib64/libaunit.so.24.0.0]
        0x7fe0ca13e4e1
        0x7fe0ca13e6a5
        0x7fe0ca143ae8
        0x7fe0ca143ae8
        0x7fe0ca13e18d
        [./exe/aunit_harness]
        0x406c55 Aunit_Harness.Harness at aunit-run.adb:87
        0x406cba Aunit_Harness at aunit_harness.adb:26
        0x4075d4 Main at b__aunit_harness.adb:503
        [/usr/lib64/libc.so.6]
        0x7fe0c9a46148
        0x7fe0c9a46209
        [./exe/aunit_harness]
        0x406b13 _start at ???
        0xfffffffffffffffe


Total Tests Run:   15
Successful Tests:  14
Failed Assertions: 0
Unexpected Errors: 1

OK (test_case) Test routines registration (in 0.000002 sec.)

Total Tests Run:   1
Successful Tests:  1
Failed Assertions: 0
Unexpected Errors: 0
```
